core/model/configuration.py

```diff
@@ -1140,6 +1140,11 @@ def __get__(
                 logging.getLogger(self.__class__.__name__).error(
                     f"Could not typecast {self.label} value '{setting_value}'. Falling back to default"
                 )
+                if setting_value != "":
+                    # A non-empty value is a "bad" value, and should raise an exception
+                    raise CannotLoadConfiguration(
+                        f"Could not typecast {self.label} value '{setting_value}'. Falling back to default"
```

Comment:
I think we need to remove the last sentence here, since we’re raising an exception, rather than falling back to the default.

Would also suggest changing “…typecast…” to “…convert…” since that’s what we’re really doing here. Same suggestion applies to the log message at ll. 1140-1142.

Reply:
Brain fog. Why would I say we're falling back to default when all we're doing is falling :dizzy:

Reply:
I removed the log, the exception will cause a giant log anyways